### PR TITLE
MINOR: update Streams docs for KIP-123

### PR DESCRIPTION
**Pull request by mjsax**

### CODE REVIEW

**Comment by mjsax** [comment]
Call for review @bbejeck @miguno @dguy @guozhangwang (that's for 0.11)

File: docs/streams.html [hunk 1]
```diff
@@ -808,6 +808,12 @@
         <h2><a id="streams_upgrade_and_api" href="#streams_upgrade_and_api">Upgrade Guide and API Changes</a></h2>
 
         <p>
+        If you want to upgrade from 0.10.2.x to 0.11.0 you don't need to do any code changes as the API is fully backward compatible.
```
Comment (guozhangwang) [inline docs/streams.html]: Ping @ewencp @ijuma to confirm backward compatibility across all Kafka modules.

**Review by guozhangwang — COMMENTED** [review]
LGTM.

File: docs/streams.html (hunk 1 above)
Comment (ijuma) [reply docs/streams.html]: We should mention that this guarantee doesn't apply to deprecated (we removed a few) and internal APIs (we change those regularly).

**Review by bbejeck — APPROVED** [review]

File: docs/streams.html (hunk 1 above)
Comment (mjsax) [reply docs/streams.html]: Streams did not remove any deprecated APIs AFAIK -- if I missed one, please point it out as we should explicitly list what was removed.

And for internal APIs -- well, they are internal, no reason to talk about them IMHO.

**Comment by bbejeck** [comment]
LGTM

File: docs/streams.html (hunk 1 above)
Comment (ijuma) [reply docs/streams.html]: @mjsax I was responding to @guozhangwang's question about `all Kafka modules`. Given that internal APIs are in the classpath and there is no warning when people use it (and from what I've seen people _do_ use them), there's a case to be made that it's worth mentioning it. One way to put it is `reality` versus `theory`. I'll leave it to you all to decide though.

Comment (ewencp) [reply docs/streams.html]: I don't feel strongly about including the caveat, but I may expect too much of users -- I like to think the ones that are using internal APIs realize it. You could just put `public` in front of `APIs` and that seems plenty good to me.

For Connect, it is true that 0.11.0.0 is fully compatible with 0.10.2.0.

**Comment by mjsax** [comment]
Rebased and updated.

**Comment by guozhangwang** [comment]
LGTM. Merged to trunk and 0.11.0.

**Pull request closed without merging**